torchao/quantization/quant_api.py

```diff
@@ -1489,16 +1497,10 @@ class Float8WeightOnlyConfig(AOBaseConfig):
 
 
 def _float8_weight_only_quant_tensor(weight, config):
-    from torchao.dtypes import to_affine_quantized_floatx
+    float8_tensor_kwargs = None
 
-    block_size = tuple([1 for _ in range(weight.dim() - 1)] + [weight.shape[-1]])
-    new_weight = to_affine_quantized_floatx(
-        input_float=weight,
-        block_size=block_size,
-        target_dtype=config.weight_dtype,
-        scale_dtype=None,
-        _layout=Float8Layout(mm_config=None),
-    )
+    float8_tensor_kwargs = QuantizeTensorToFloat8Kwargs(config.weight_dtype, PerRow())
+    new_weight = _choose_quant_func_and_quantize_tensor(weight, float8_tensor_kwargs)
```

Comment:
I think this should instead call `Float8Tensor.to_float8` with the actual args, and not using the layer of indirection.  It will be easier to follow.  Would be good to keep the layer of indirection only for activations, where it is actually needed.

Reply:
OK sounds good

Reply:
actually this change is going to make `_choose_quant_func_and_quantize_tensor` more complicated as well, just FYI.